### PR TITLE
Redirect on plans page for P2-non-hubs

### PR DESCRIPTION
**Pull request by naxoc**
#### Changes proposed in this Pull Request

* For P2s – only hubs can have a paid plan. When hitting a plans page on a P2 site that belongs to a hub, we redirect to the _hub's_ plans page.

Note that this PR also contains the change in #52722 for ease of testing. 

The PR also needs `D61198-code`. That should be why it's failing tests here.

#### Testing instructions

TBA

*

### CODE REVIEW

File: client/state/selectors/get-p2-hub-blog-id.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { get } from 'lodash';
+import isSiteWPForTeams from 'calypso/state/selectors/is-site-wpforteams';
+
+/**
+ * Returns the P2 hub blog id for the site. If the site is a hub the id will be
+ * the same as the site id.
+ *
+ * @param {object} state Global state tree
+ * @param {number} siteId the site ID
+ * @returns {number} % of file uploaded so far
```
Comment (lamosty) [inline client/state/selectors/get-p2-hub-blog-id.js]: Typo

File: client/state/selectors/get-p2-hub-blog-id.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ */
+export default function getP2HubBlogId( state, siteId ) {
+	if ( ! isSiteWPForTeams( state, siteId ) ) {
+		return 0;
```
Comment (lamosty) [inline client/state/selectors/get-p2-hub-blog-id.js]: I think we can return `null` here. That's usually what we return from selectors on error in Calypso.

File: client/state/selectors/is-site-wpforteams-hub.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import isSiteWPForTeams from 'calypso/state/selectors/is-site-wpforteams';
+
+/**
+ * Returns true if site is a WP for Teams hub, false if not and null if unknown.
+ *
+ * @param  {object}   state  Global state tree
+ * @param  {number}   siteId Site ID
+ * @returns {?boolean}        Whether site is a WP for Teams site hub
+ */
+export default function isSiteWPForTeamsHub( state, siteId ) {
```
Comment (lamosty) [inline client/state/selectors/is-site-wpforteams-hub.js]: I suppose we could start using the P2 naming instead of wp for teams for the new code.

**Review by lamosty — COMMENTED** [review]
Nice work!

Left a few smaller comments.

File: client/my-sites/controller.js [hunk 4]
```diff
@@ -591,6 +593,32 @@ export function wpForTeamsP2PlusNotSupportedRedirect( context, next ) {
...
 }
 
+/**
+ * For P2s, only hubs can have a plan. If we are on P2 a site that is a site under
+ * a hub, we redirect the hub's plans page.
+ *
+ * @param {object} context -- Middleware context
+ * @param {Function} next -- Call next middleware in chain
+ */
+export function wpForTeamsRedirectToHubPlans( context, next ) {
```
Comment (lamosty) [inline client/my-sites/controller.js]: Could we use the P2 naming here too?

Comment (naxoc) [reply client/my-sites/controller.js]: I changed the naming 👍

File: client/state/selectors/get-p2-hub-blog-id.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * @param {object} state Global state tree
+ * @param {number} siteId the site ID
+ * @returns {?number} Blog ID of the site's hub if it is a P2 – otherwise null.
+ */
+export default function getP2HubBlogId( state, siteId ) {
+	if ( ! isSiteWPForTeams( state, siteId ) ) {
+		return null;
+	}
+	return get( state, [ 'sites', 'items', siteId, 'options', 'p2_hub_blog_id' ], 0 );
```
Comment (lamosty) [inline client/state/selectors/get-p2-hub-blog-id.js]: Could you update the `0` to `null` as well since the default bad case is `null`?

Comment (naxoc) [reply client/state/selectors/get-p2-hub-blog-id.js]: Good catch! 👍

**Review by lamosty — APPROVED** [review]
Thanks for addressing all the feedback!

I've tested:
- going from a hub's plans page to a non-hub one;
- going from a non-workspace p2 plans page to a non-hub workspace one;
- going from a hub's plans page to a wpcom site;
- going from a wpcom site's plans page to a non-hub workspace one;

and all seems to be working as expected.

LGTM! 🚢

**Pull request merged**